tests/integration/test_master_coordinator.py

```diff
@@ -44,82 +40,82 @@ def has_master(mc: MasterCoordinator) -> bool:
 
 @pytest.mark.timeout(60)  # Github workflows need a bit of extra time
 @pytest.mark.parametrize("strategy", ["lowest", "highest"])
-def test_master_selection(kafka_servers: KafkaServers, strategy: str) -> None:
+def test_master_selection(port_range: PortRangeInclusive, kafka_servers: KafkaServers, strategy: str) -> None:
     # Use random port to allow for parallel runs.
-    port1 = get_random_port(port_range=TESTS_PORT_RANGE, blacklist=[])
-    port2 = get_random_port(port_range=TESTS_PORT_RANGE, blacklist=[port1])
-    port_aa, port_bb = sorted((port1, port2))
-    client_id_aa = new_random_name("master_selection_aa_")
-    client_id_bb = new_random_name("master_selection_bb_")
-    group_id = new_random_name("group_id")
-
-    config_aa = set_config_defaults(
-        {
-            "advertised_hostname": "127.0.0.1",
-            "bootstrap_uri": kafka_servers.bootstrap_servers,
-            "client_id": client_id_aa,
-            "group_id": group_id,
-            "port": port_aa,
-            "master_election_strategy": strategy,
-        }
-    )
-    config_bb = set_config_defaults(
-        {
-            "advertised_hostname": "127.0.0.1",
-            "bootstrap_uri": kafka_servers.bootstrap_servers,
-            "client_id": client_id_bb,
-            "group_id": group_id,
-            "port": port_bb,
-            "master_election_strategy": strategy,
-        }
-    )
-
-    with closing(init_admin(config_aa)) as mc_aa, closing(init_admin(config_bb)) as mc_bb:
-        if strategy == "lowest":
-            master = mc_aa
-            slave = mc_bb
-        else:
-            master = mc_bb
-            slave = mc_aa
-
-        # Wait for the election to happen
-        while not is_master(master):
-            time.sleep(0.3)
-
-        while not has_master(slave):
-            time.sleep(0.3)
-
-        # Make sure the end configuration is as expected
-        master_url = f'http://{master.config["host"]}:{master.config["port"]}'
-        assert master.sc.election_strategy == strategy
-        assert slave.sc.election_strategy == strategy
-        assert master.sc.master_url == master_url
-        assert slave.sc.master_url == master_url
-
-
-def test_no_eligible_master(kafka_servers: KafkaServers) -> None:
+    with port_range.allocate_port() as port1, port_range.allocate_port() as port2:
```

Comment:
I didn't change the test, only added the context manager for the port allocation.